x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java

```diff
@@ -950,59 +967,29 @@ public enum AuthenticationType {
         INTERNAL
     }
 
-    // Package private for testing
-    static class RunAsUser extends User {
-        final User authenticatingUser;
-
-        RunAsUser(User effectiveUser, User authenticatingUser) {
-            super(
-                effectiveUser.principal(),
-                effectiveUser.roles(),
-                effectiveUser.fullName(),
-                effectiveUser.email(),
-                effectiveUser.metadata(),
-                effectiveUser.enabled()
-            );
-            this.authenticatingUser = Objects.requireNonNull(authenticatingUser);
-        }
+    public static class AuthenticationSerializationHelper {
```

Comment:
This class now only deals with regular User object. It does not need to stay inside the Authentication class anymore. But I kept it here to avoid more changes. Besides the usages by Authentication, it is only used by GetUserResponse which should ever just read/write regular User objects.